src/mono/mono/metadata/loader-internals.h

```diff
@@ -62,6 +72,53 @@ struct _MonoAssemblyLoadContext {
 };
 #endif /* ENABLE_NETCORE */
 
+struct _MonoMemoryManager {
+	// Whether the MemoryManager can be unloaded on netcore; should only be set at creation
+	gboolean collectible;
+	// Whether this is a singleton or generic MemoryManager
+	gboolean is_generic;
+	// Whether the MemoryManager is in the process of being freed
+	gboolean freeing;
+
+	// Entries moved over from the domain:
+
+	// If taking this with the loader lock, always take this second
+	// On legacy, this does _not_ protect mp/code_mp, which are covered by the domain lock
+	MonoCoopMutex lock;
+
```

Comment:
Does this have to be a coop mutex ?

Reply:
I think so? Some of the data structures it protects are coop-aware.